src/boost_histogram/_internal/hist.py

```diff
@@ -885,6 +890,30 @@ def __getitem__(self: H, index: IndexingExpr) -> H | float | Accumulator:
             logger.debug("Reduce actions are all empty, just making a copy")
             reduced = copy.copy(self._hist)
 
+        # bin re-grouping
```

Comment:
I _think_ this should be moved to where "merge" is processed? I think it's safe to do this in the loop as long as it doesn't add or remove axes (which it doesn't, only possibly swaps an axis)